code/datums/gamemode/factions/faction.dm

```diff
@@ -116,6 +132,74 @@
 /datum/faction/proc/check_win()
 	return
 
+//updating every icons at the same time allows their animate() to be sync'd, so we can alternate the one on top without any additional proc calls.
+/proc/update_faction_icons()
+	if (!ticker || !ticker.mode)
+		return
+
+	var/offset = 0
+	for (var/datum/faction/F in ticker.mode.factions)
+		if (F.hud_icons.len)
+			F.update_hud_icons(offset)
+			offset++
+
+#define HUDICON_BLINKDURATION 10//the smaller, the faster icons swap to one another
+/datum/faction/proc/update_hud_icons(var/offset = 0)
+	//lets ignore this proc if our faction has no icons (for factions where we don't want its members to know each others by default)
+	if (!hud_icons.len)
+		return
+
+	var/factions_with_icons = 0
+	for (var/datum/faction/F in ticker.mode.factions)
+		if (F.hud_icons.len)
+			factions_with_icons++
+
+	spawn(0)
+		//let's remove every icons
+		for(var/datum/role/R in members)
+			if(R.antag && R.antag.current && R.antag.current.client)
+				for(var/image/I in R.antag.current.client.images)
+					if(I.icon_state in hud_icons)
+						R.antag.current.client.images -= I
+
+		//then re-add them
+		for(var/datum/role/R in members)
+			if(R.antag && R.antag.current && R.antag.current.client)
+				for(var/datum/role/R_target in members)
+					if(R_target.antag && R_target.antag.current)
```

Comment:
None of these procs sleep or spawn. What's stopping you from doing it all in one go ?